### DISCUSSION TITLE
Timers / Command Queue

### DISCUSSION THREAD

**Discussion by DuncanC123** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#581] [General]
Hello

I have it all setup and generally working great, thank you.

The only thing i cant work out is why the command queue doesnt seem to be executing properly unless i am standing next to the tub.
I simply want it to be set to a lower temp at 4pm everyday when elec is more expensive and back to the normal temp at 7pm when cost comes down again.
I have put in the commands, see screenshot attached, but when i get home which is between those times it is still at the high temp.
When testing it while at home it works though.

Only thing i can think is does the schedule get stored on my device i.e. phone, so it can only execute when i am on the network?
Hopefully there is a way to get it to change when i am not home, that was the main reason i wanted to make this controller.

Thanks in advance

<img width="434" alt="Screenshot 2023-07-31 at 18 58 17" src="https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/141063779/850aec5b-7b3c-47bc-82af-99d25dd7dd51">

**Comment by visualapproach** [comment]
Do you have a Quantum hot tub? It only exists when you look at it 😄.
The command queue is stored locally in the ESP and not in your phone. Only thing I can think of is that the time is not set correctly. You can check it on the main web page.

**Reply by DuncanC123** [reply]
Must be Quantum!!

I have checked the times and dont think its that. I can set it to change temp in 1 min (correct time) and then watch it change.
Its only been running 2 days but both days i have checked at some point between 4-7pm and it was still at 38deg.
I tapped the temp button on the unit so i know the target temp, not simply confusing it with the current temp.

Good to hear the command is saved locally so it should change temp as i planned.
Is there anything else it could be?

Also side note. I have turned on notifications, set it to 32s, added a command for 1-2min ahead and it changes temp but no noise to notify?

**Reply by visualapproach** [reply]
> Must be Quantum!!
> 
> 
> 
> I have checked the times and dont think its that. I can set it to change temp in 1 min (correct time) and then watch it change.
> 
> Its only been running 2 days but both days i have checked at some point between 4-7pm and it was still at 38deg.
> 
> I tapped the temp button on the unit so i know the target temp, not simply confusing it with the current temp.
> 
> 
> 
> Good to hear the command is saved locally so it should change temp as i planned.
> 
> Is there anything else it could be?


It appears so. We only need to find out what. See if you can spot a pattern. Is it delayed, too early, not executing at all, or not working only when your brother in law visits, and so on


> Also side note. I have turned on notifications, set it to 32s, added a command for 1-2min ahead and it changes temp but no noise to notify?

It depends. Sound and display text is only possible on 6-wire setups. And audio needs to be enabled ofc

**Reply by DuncanC123** [reply]
It worked today!
No idea what did it, maybe just playing with values and making it re-save, but it now changes temp when i'm away and has sound notifications.
Thanks for trying to help.

Now on to the big hurdle of researching mqtt and if i can use HA to control the tub based on the Octopus energy api...

**Reply by visualapproach** [reply]
Yes connecting to HA should be easy. Just install mosquitto and setup a user. Then config the ESP to use the ip to the HA computer and the user you just created.
